### PR TITLE
added ISI, CCT-cold, and CCT-hot

### PR DESCRIPTION
**Pull request by jkl071**

### CODE REVIEW

File: CCT_cold/config.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[
+    {
+        "name": “Columbia Card Task Cold Version“,
+        "run": [
```
Comment (vsoch) [inline CCT_cold/config.json]: See the red here? The " character you are using is not the right one (what are you using for a text editor?) Try copy pasting from a file that you know does not have this invalid character.

File: CCT_cold/config.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+                 "experiment.js",
+                 "style.css"
+               ],
+        "tag": “Columbia_card_task_cold”,
+        "cognitive_atlas_task_id": "trm_5667492c555b7",
```
Comment (vsoch) [inline CCT_cold/config.json]: ditto here

File: CCT_cold/experiment.js [hunk 3]
```diff
@@ -0,0 +1,665 @@
...
+		gameState = appendTextAfter(gameState, 'Current Round: ', 0) //make a function to find cumulative score for this round
+		gameState = appendTextAfter(gameState, '# of Loss Cards: ', lossProb)
+		gameState = appendTextAfter(gameState, 'Gain Amount: ', gainAmt)
+	return gameState
+}
+
+
+var instructFunction = function(){
+	document.getElementById("c1").src='/images/chosen.png';
+	document.getElementById("c17").src='/images/chosen.png';
```
Comment (vsoch) [inline CCT_cold/experiment.js]: This hard-coded images path likely will not work in the infrastructure - the path must be to /static/experiments/[experimentname]/images

File: CCT_cold/experiment.js [hunk 4]
```diff
@@ -0,0 +1,665 @@
...
+  "<button type='button' class = select-button2 id = 24  onclick = chooseButton1(this.id)>24</button>"+
+  "<button type='button' class = select-button2 id = 25  onclick = chooseButton1(this.id)>25</button>"+
+  "<button type='button' class = select-button2 id = 26  onclick = chooseButton1(this.id)>26</button>"+
+  "<button type='button' class = select-button2 id = 27  onclick = chooseButton1(this.id)>27</button>"+
+  "<button type='button' class = select-button2 id = 28  onclick = chooseButton1(this.id)>28</button>"+
+  "<button type='button' class = select-button2 id = 29  onclick = chooseButton1(this.id)>29</button>"+
+  "<button type='button' class = select-button2 id = 30  onclick = chooseButton1(this.id)>30</button>"+
+  "<button type='button' class = select-button2 id = 31  onclick = chooseButton1(this.id)>31</button>"+
+  "<button type='button' class = select-button2 id = 32  onclick = chooseButton1(this.id)>32</button></div>"+
+
```
Comment (vsoch) [inline CCT_cold/experiment.js]: You might consider adding elements with Jquery or the like (eg a loop) to produce these elements, vs. hard coding them all.

File: CCT_hot/config.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+[
+    {
+        "name": “Columbia Card Task Hot Version“,
+        "run": [
```
Comment (vsoch) [inline CCT_hot/config.json]: same issue here with characters

File: CCT_hot/config.json [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+                 "experiment.js",
+                 "style.css"
+               ],
+        "tag": “Columbia_card_task_hot”,
+        "cognitive_atlas_task_id": "trm_5667492c555b7",
```
Comment (vsoch) [inline CCT_hot/config.json]: tags must be all lowercase

File: ISI/config.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
+[
+    {
+        "name": “Information Sampling Task,
+        "run": [
+                 "experiment.js",
+                 "style.css"
+               ],
+        "tag": “ISI”,
+        "cognitive_atlas_task_id": "trm_50f73d557c967",
```
Comment (vsoch) [inline ISI/config.json]: lowercase and illegal characters here

File: ISI/config.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
+[
+    {
+        "name": “Information Sampling Task,
+        "run": [
```
Comment (vsoch) [inline ISI/config.json]: illegal character and missing a closing parens

File: ISI/experiment.js [hunk 9]
```diff
@@ -0,0 +1,691 @@
...
+ISI_experiment.push(practice_chunk);
+ISI_experiment.push(rewardFW_block);
+ISI_experiment.push(reset_block);
+
+ISI_experiment.push(practice_chunk);
+ISI_experiment.push(rewardFW_block);
+ISI_experiment.push(reset_block);
+
+ISI_experiment.push(practice_chunk);
+ISI_experiment.push(rewardFW_block);
```
Comment (vsoch) [inline ISI/experiment.js]: when you change the task names to lowercase, be sure to change these variables as well to lowercase, eg `isi_experiment`

File: CCT_cold/config.json [hunk 10]
```diff
@@ -0,0 +1,25 @@
+[
+    {
+        "name": "Columbia Card Task Cold Version",
+        "run": [
+                 "experiment.js",
+                 "style.css"
+               ],
+        "tag": "columbia_card_task_cold",
```
Comment (vsoch) [inline CCT_cold/config.json]: here the tag is "columbia_card_task_cold" but the experiment has "CCT_cold"

Comment (vsoch) [inline CCT_cold/config.json]: to be more specific, the tag variable must correspond to the folder name, which must be lowercase, no special characters. etc

File: CCT_cold/experiment.js [hunk 11]
```diff
@@ -0,0 +1,665 @@
...
+			return true
+		}
+    }
+}
+
+var test_block = {
+  type: 'single-stim',
+  stimuli: getRound,
+  is_html: true,
+  data: {exp_id: "CCT"},
```
Comment (vsoch) [inline CCT_cold/experiment.js]: same here - the experiment id is the tag, is the folder name.

File: CCT_hot/experiment.js [hunk 12]
```diff
@@ -0,0 +1,818 @@
...
+	document.getElementById(clicked_id).src='/static/experiments/CCT_hot/images/chosen.png';
+}
+
+var instructFunction = function(){
+	document.getElementById("1").src='/static/experiments/CCT_hot/images/chosen.png';
+	document.getElementById("17").src='/static/experiments/CCT_hot/images/chosen.png';
+	document.getElementById("18").src='/static/experiments/CCT_hot/images/chosen.png';
+	document.getElementById("15").src='/static/experiments/CCT_hot/images/chosen.png';
+	document.getElementById("27").src='/static/experiments/CCT_hot/images/chosen.png';
+	document.getElementById("31").src='/static/experiments/CCT_hot/images/chosen.png';
```
Comment (vsoch) [inline CCT_hot/experiment.js]: ditto here

File: CCT_hot/experiment.js [hunk 13]
```diff
@@ -0,0 +1,818 @@
...
+  timing_post_trial: 1000
+};
+
+
+
+var practice_block = {
+  type: 'single-stim',
+  stimuli: getRound,
+  is_html: true,
+  data: {exp_id: "CCT", trial: 'test'},
```
Comment (vsoch) [inline CCT_hot/experiment.js]: ditto, and the two different tasks need to have different names

File: ISI/experiment.js [hunk 14]
```diff
@@ -0,0 +1,691 @@
...
+  if(roundOver == 0){ // start of the round
+		gameState=gameSetup
+			gameState=appendTextAfter2(gameState, "id = '26'", " src='/static/experiments/ISI/images/"+whichLargeColor1+".png'")		
+			gameState=appendTextAfter2(gameState, "id = '27'", " src='/static/experiments/ISI/images/"+whichLargeColor2+".png'")		
+
+		return gameState
+  }else if(roundOver == 1){
+  			gameState=gameSetup
+  			gameState=appendTextAfter2(gameState, "id = '26'", " src='/static/experiments/ISI/images/"+whichLargeColor1+".png'")		
+			gameState=appendTextAfter2(gameState, "id = '27'", " src='/static/experiments/ISI/images/"+whichLargeColor2+".png'")
```
Comment (vsoch) [inline ISI/experiment.js]: the folder names should be the same as the tag, not capitalized

File: ISI/experiment.js [hunk 15]
```diff
@@ -0,0 +1,691 @@
...
+  text: '<div class = centerbox><p class = block-text>We will show you a practice trial.  Press <strong>enter</strong> to begin.</p></div>',
+  cont_key: 13,
+  timing_post_trial: 1000
+};
+
+var subjectPracticeBlock ={
+  type: 'single-stim',
+  stimuli: instructionsSetup,
+  is_html: true,
+  data: {exp_id: "ISI", trial_id: "test"},
```
Comment (vsoch) [inline ISI/experiment.js]: same with these experiment ids

File: CCT_cold/config.json (hunk 10 above)
Comment (jkl071) [inline CCT_cold/config.json]: Hmm, my CCT-cold.js file says that the experiment name is columbia_card_task_cold...

Should I be changing it somewhere else?

Comment (vsoch) [inline CCT_cold/config.json]: The experiment "tag" must be equivalent to the folder name, which must be equivalent to the exp_id in Javascript. Currently we have:

- experiment tag: "columbia_card_task_cold"
- folder name: "CCT_cold"
- paths embedded in script: "CCT_cold"
- exp_id variable: "CCT"

all of these things must be the same, I would recommend changing to the first as it is more descriptive, and follows the standard of lowercase without special characters. The experiment will not validate in its current form.

**Comment by jkl071** [comment]
changing to lowercase

**Comment by vsoch** [comment]
Good work! What you can do now is use the output from CircleCI to debug your Javascript (scroll down to the section in RED to see the jshint output that will tell you how to fix your current code. And one small detail - please change the experiment tag / folder name / references in the experiment.js for the "isi" experiment to something more meaningful - I changed a bunch of the old names this weekend toward this goal.

**Comment by vsoch** [comment]
You can click the "details" button to see the continuous integration running - let me know if you have questions about it, because it's sort of weird looking the first time you see it. It will test two things - first code syntax with jshint, and then validation of the config.json. We don't yet have any kind of "robot" that actually runs through the tasks (and checks for runtime errors) but hopefully this is something I can develop.

**Comment by vsoch** [comment]
looks like you didn't delete isi from the repo, if it's the same thing as "information_sampling_task."  The checks will not pass until you address the JSHint errors.

**Comment by IanEisenberg** [comment]
I am seeing some warnings and "no cognitive atlas id" as an error, but other than that I think we can merge.  We are going to have to ignore the cognitive atlas id errors for now.

**Comment by vsoch** [comment]
It's really trivial to add a cognitive_atlas_task_id to the config.json. It's required and the tasks won't be deployable without it, so I think it makes sense to add.

**Comment by IanEisenberg** [comment]
What if we don't have one for the particular task? I suppose we could make
a task...

On Tuesday, January 26, 2016, Vanessa Sochat <notifications@github.com>
wrote:

> It's really trivial to add a cognitive_atlas_task_id to the config.json.
> It's required and the tasks won't be deployable without it, so I think it
> makes sense to add.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/expfactory/expfactory-experiments/pull/46#issuecomment-175371961>
> .
>


-- 
Best,
Ian

**Comment by vsoch** [comment]
Correct as usual, King Friday!

If you want to send me info stuffs, I can make them and send you back the ids.

**Comment by jkl071** [comment]
Hey @vsoch, I am not quite sure why my error is still coming up.  Could you see if you find something funny?

**Comment by vsoch** [comment]
Sure will take a look, one sec!

**Comment by vsoch** [comment]
Do you see where it says:

      behavioral_approach_behavioral_inhibition_systems: config.json must be defined for field cognitive_atlas_task_id

**Comment by vsoch** [comment]
remember, look for red, and then read the messages, they tell you what is wrong! If you want to validate your entire folder before pushing again (on your local machine) you can cd to it and do:

      python -c "from expfactory.tests import validate_experiment_directories; validate_experiment_directories('.')"

As long as you have the latest version of expfactory-python installed

**Comment by vsoch** [comment]
when you do python "-c" the c argument is telling python to run the next string as a command (eg, a line within a script)

**Comment by vsoch** [comment]
Also, I don't think you've rebased. We don't have a task called "rng" or "prp" anymore.

**Comment by IanEisenberg** [comment]
Yeah,  it seems like the last edit fixed the "stim" issue. Just rebase and add a cognitive atlas id (/make a cognitive atlas entry) for the behavioral_approach_behavioral_inhibition questionnaire and it should be good!

**Pull request merged**